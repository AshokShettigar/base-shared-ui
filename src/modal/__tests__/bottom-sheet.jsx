import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import BottomSheet, { BottomSheetHOC } from 'platform-common-ui/modal/bottom-sheet';

const mockStore = configureStore();

describe('modal/bottom-sheet', function() {
    const ModalComponent = () => <div>modal</div>;
    const TestComponent = BottomSheetHOC(ModalComponent);
    const BottomSheetTest = BottomSheet(ModalComponent);
    const onClosePopup = jest.fn();

    it('should render the bottom-sheet component', function() {
        const wrapper = mount(
            <Provider store={mockStore()}>
                <BottomSheetTest
                    id="defaultBottomSheet"
                    onClosePopup={onClosePopup}
                />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the bottom-sheet HOC component', function() {
        const wrapper = shallow(
            <TestComponent
                id="defaultBottomSheet"
                title="Title"
                onClosePopup={onClosePopup}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the bottom-sheet HOC with layoverClassName', function() {
        const wrapper = shallow(
            <TestComponent
                id="defaultBottomSheet"
                title="Title"
                layoverClassName="layover-classname"
                onClosePopup={onClosePopup}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the bottom-sheet HOC with popupClassName', function() {
        const wrapper = shallow(
            <TestComponent
                id="defaultBottomSheet"
                title="Title"
                popupClassName="popup-classname"
                onClosePopup={onClosePopup}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the bottom-sheet HOC with width prop', function() {
        const wrapper = shallow(
            <TestComponent
                id="defaultBottomSheet"
                title="Title"
                width="650px"
                onClosePopup={onClosePopup}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the bottom-sheet HOC with height prop', function() {
        const wrapper = shallow(
            <TestComponent
                id="defaultBottomSheet"
                title="Title"
                height="400px"
                onClosePopup={onClosePopup}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    // it('should render the bottom-sheet HOC with onClose function', function() {
    //     const wrapper = shallow(
    //         <TestComponent
    //             id="defaultBottomSheet"
    //             title="Title"
    //             onClosePopup={onClosePopup}
    //         />
    //     );
    //     wrapper.find('.cn-bottom-sheet__navbar__close').simulate('click');
    //     expect(onClosePopup).toHaveBeenCalledTimes(1);
    //     expect(onClosePopup).toHaveBeenCalledWith('defaultBottomSheet');
    // });
});
