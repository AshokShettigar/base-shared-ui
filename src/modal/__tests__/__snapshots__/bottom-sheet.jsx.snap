// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal/bottom-sheet should render the bottom-sheet HOC component 1`] = `
<BaseModal
  height="450px"
  id="defaultBottomSheet"
  onClosePopup={[Function]}
  title="Title"
  width="600px"
>
  <ModalComponent
    id="defaultBottomSheet"
    onClosePopup={[Function]}
    title="Title"
  />
</BaseModal>
`;

exports[`modal/bottom-sheet should render the bottom-sheet HOC with height prop 1`] = `
<BaseModal
  height="400px"
  id="defaultBottomSheet"
  onClosePopup={[Function]}
  title="Title"
  width="600px"
>
  <ModalComponent
    height="400px"
    id="defaultBottomSheet"
    onClosePopup={[Function]}
    title="Title"
  />
</BaseModal>
`;

exports[`modal/bottom-sheet should render the bottom-sheet HOC with layoverClassName 1`] = `
<BaseModal
  height="450px"
  id="defaultBottomSheet"
  onClosePopup={[Function]}
  title="Title"
  width="600px"
>
  <ModalComponent
    id="defaultBottomSheet"
    layoverClassName="layover-classname"
    onClosePopup={[Function]}
    title="Title"
  />
</BaseModal>
`;

exports[`modal/bottom-sheet should render the bottom-sheet HOC with popupClassName 1`] = `
<BaseModal
  height="450px"
  id="defaultBottomSheet"
  onClosePopup={[Function]}
  title="Title"
  width="600px"
>
  <ModalComponent
    id="defaultBottomSheet"
    onClosePopup={[Function]}
    popupClassName="popup-classname"
    title="Title"
  />
</BaseModal>
`;

exports[`modal/bottom-sheet should render the bottom-sheet HOC with width prop 1`] = `
<BaseModal
  height="450px"
  id="defaultBottomSheet"
  onClosePopup={[Function]}
  title="Title"
  width="650px"
>
  <ModalComponent
    id="defaultBottomSheet"
    onClosePopup={[Function]}
    title="Title"
    width="650px"
  />
</BaseModal>
`;

exports[`modal/bottom-sheet should render the bottom-sheet component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BottomSheetModal
    id="defaultBottomSheet"
    layoverClassName="cn-modal-layover"
    onClosePopup={[Function]}
    popupClassName="cn-modal cn-bottom-sheet"
  >
    <HigherOrderPopupComponent
      id="defaultBottomSheet"
      layoverClassName="cn-modal-layover"
      onClosePopup={[Function]}
      popupClassName="cn-modal cn-bottom-sheet"
      updatePopupProps={[Function]}
    />
  </BottomSheetModal>
</Provider>
`;
